vm/src/obj/objint.rs

```diff
@@ -736,21 +739,22 @@ fn int_new(cls: PyClassRef, options: IntOptions, vm: &VirtualMachine) -> PyResul
 }
 
 // Casting function:
-pub fn to_int(vm: &VirtualMachine, obj: &PyObjectRef, mut base: u32) -> PyResult<BigInt> {
-    if base == 0 {
-        base = 10
-    } else if base < 2 || base > 36 {
+pub fn to_int(vm: &VirtualMachine, obj: &PyObjectRef, base: u32) -> PyResult<BigInt> {
+    if base != 0 && (base < 2 || base > 36) {
         return Err(vm.new_value_error("int() base must be >= 2 and <= 36, or 0".to_string()));
     }
 
     match_class!(obj.clone(),
-        s @ PyString => {
-            i32::from_str_radix(s.as_str().trim(), base)
-                .map(BigInt::from)
-                .map_err(|_|vm.new_value_error(format!(
-                    "invalid literal for int() with base {}: '{}'",
-                    base, s
-                )))
+        string @ PyString => {
+            let s = string.value.as_str().trim();
+            str_to_int(vm, s, base)
+        },
+        bytes @ PyBytes => {
+            let bytes = bytes.get_value();
+            let s = std::str::from_utf8(bytes)
+                .map(|s| s.trim())
+                .map_err(|_| invalid_literal(vm, &bytes.iter().map(|&c| c as char).collect::<String>(), base))?;
```

Comment:
You might want to raise a TypeError here and use the rust error message like this:
```rust
.map_err(|e| vm.new_type_error(format!("utf8 decode error: {}", e)))?;
```

Reply:
I changed it to ```TypeError``` and added some tests covering that branch - but it looks that ```cpython``` uses ```ValueError``` right here.

Should we switch to ```TypeError``` or rather stay with ```cpython``` behavior?

Reply:
Ah, okay, a `ValueError` is a better idea. I meant more the error message which is returned from the `std::from_utf8` function, we might as well use it, to indicate in a better way what went wrong.

```rust
.map_err(|e| vm.new_value_error(format!("utf8 decode error: {}", e)))?;
```

Reply:
Ok, so I'll return ```ValueError``` and apply Your suggestion :)